docs/page-objects/input-validation.rst

```diff
@@ -0,0 +1,80 @@
+.. _input-validation:
+
+================
+Input validation
+================
+
+Sometimes the data that your page object receives as input may be invalid.
+
+You can define a ``validate_input`` method in a page object class to check its
+input data and determine how to handle invalid input. ``validate_input``
+is a synchronous method that expects no parameters, and its outcome may be any
+of the following:
+
+-   Return ``None``, indicating that the input is valid.
+
+.. _retries-input:
+
+-   Raise :exc:`~web_poet.exceptions.Retry`, indicating that the input
+    looks like the result of a temporary issue, and that trying to fetch
+    similar input again may result in valid input.
+
+    See also :ref:`retries-additional-requests`.
+
+-   Raise :exc:`~web_poet.exceptions.UseFallback`, indicating that the
+    page object does not support the input, and that an alternative parsing
+    implementation should be tried instead.
+
+    For example, imagine you have a page object for website commerce.example,
+    and that commerce.example is built with Shopify. You could have a generic
+    page object for products of Shopify-based websites, ``ShopifyProductPage``,
+    and a more specific page object for commerce.example,
+    ``EcommerceExampleProductPage``. If ``EcommerceExampleProductPage`` cannot
+    parse a product page, but it looks like it might be a valid product page,
+    you would raise :exc:`~web_poet.exceptions.UseFallback` to try to
+    parse the same product page with ``ShopifyProductPage``, in case it works.
+
+    .. note:: web-poet does not yet allow defining the alternative parsing
+              implementation to use as fallback, so raising this exception has
+              no effect with the current version of web-poet.
```

Comment:
I think it's more that the frameworks we have (scrapy-poet and web_poet.example) don't have an implementation; I'm not sure it's web-poet's job to handle this exception.

Reply:
There is an effect already - the extraction stops, and exception is propagated; it seems this would be the same in future web-poet versions as well.

Reply:
https://github.com/scrapinghub/web-poet/pull/113/commits/8dcc57c8b847614d31579787948928e3d12615f2 ?